ports/unix/mpthreadport.c

```diff
@@ -109,6 +109,8 @@ void mp_thread_init(void) {
     sa.sa_sigaction = mp_thread_gc;
     sigemptyset(&sa.sa_mask);
     sigaction(SIGUSR1, &sa, NULL);
+    sigaction(SIGUSR2, &sa, NULL);
```

Comment:
This doesn't look right because these additional signals are not related to GC, and so shouldn't be configured to call `mp_thread_gc` when they fire.

Instead such signals should be configured by the user to do what the users needs when they fire.

Reply:
@dpgeorge Without calling `sigaction`, these signals terminate the application by default, which is not what we want.   

These signals won't really affect `mp_thread_gc` because it is guarded by `if`:
https://github.com/micropython/micropython/blob/0b86ba565c3d5093620faeb0d5abdd3ff56beffa/ports/unix/mpthreadport.c#L69

Can we call `sigaction` from python instead of from C?
Or do you think we need to define a dummy `struct sigaction` in C for these signals?

Reply:
How exactly are you using the new behaviour introduced in this PR, are you modifying the C code to deal with signals or doing it purely from Python?  What is the source of the signal events in your case?

> Can we call `sigaction` from python instead of from C?

Yes that is possible, see the `signal` module in micropython-lib.

Reply:
Ok.  
I removed the calls to `sigaction`.